### PR TITLE
Add persistent functionality

### PR DESCRIPTION
**Pull request by darkin1**
RabbitMQ has [persistent message](https://www.rabbitmq.com/persistence-conf.html)  functionality that has not been serviced. 
To use it, just add to your app/Jobs/SomeJob.php field `$persistent = true`. After that when you restart RabbitMQ message still be in a queue.

### CODE REVIEW

**Comment by makasim** [comment]
I think it would be better to always use persistent queues for AMQP\RabbitMQ transports. 

Something like this would be great: 

```php
<?php

use Interop\Amqp\AmqpMessage;

if ($message instanceof AmqpMessage) {
  $message->setDeliveryMode(AmqpMessage::DELIVERY_MODE_PERSISTENT);
}
```

**Comment by darkin1** [comment]
Updated, please check

File: src/Queue.php [hunk 1]
```diff
@@ -56,9 +56,15 @@ public function push($job, $data = '', $queue = null)
      */
     public function pushRaw($payload, $queue = null, array $options = [])
     {
+        $message = $this->psrContext->createMessage($payload);
+
+        if ($message instanceof \Interop\Amqp\Impl\AmqpMessage) {
```
Comment (makasim) [inline src/Queue.php]: could you import namespace.

Comment (darkin1) [reply src/Queue.php]: For interface `\Interop\Amqp\AmqpMessage` or implementation `Interop\Amqp\AmqpMessage`?
We can't use both or I could use an alias. Which you prefer?

Comment (makasim) [reply src/Queue.php]: use Interop\Amqp\AmqpMessage;

if ($message instanceof AmqpMessage) {

**Pull request merged**